src/main/java/lesser/brickbreaker/BrickBreakerFrame.java

```diff
@@ -35,61 +31,59 @@ public BrickBreakerFrame() {
 
         add(view);
         view.setBounds(0, 0, 800, 600);
-
-        setFocusable(true);
-        requestFocusInWindow();
-
-        initializeBricks();
-
+        setVisible(true);
         addKeyListener(new KeyAdapter() {
             @Override
             public void keyPressed(KeyEvent e) {
-                int keyCode = e.getKeyCode();
-                int paddleSpeed = paddle.getSpeed();
-
-                // Move paddle left or right
-                if (keyCode == KeyEvent.VK_LEFT) {
-                    paddle.x = Math.max(0, paddle.x - paddleSpeed);
-                } else if (keyCode == KeyEvent.VK_RIGHT) {
-                    paddle.x = Math.min(view.getWidth() - paddle.width, paddle.x + paddleSpeed);
-                }
-
-                if (keyCode == KeyEvent.VK_UP && !ballMoving) {
-                    ballMoving = true;
-                    controller.startGame();
-                }
-
-                view.repaint();
+                controller.handleKeyEvent(e.getKeyCode());
             }
         });
 
+        setFocusable(true);
+        requestFocusInWindow();
+
+        controller.initializeBricks();
+
         Timer gameTimer = new Timer(10, e -> {
-            if (ballMoving) {
+            if (!controller.isGameStopped()) {
                 controller.updateBallPosition();
             }
-            if (controller.isGameStopped()) {
-                ballMoving = false;
-            }
+
             view.repaint();
         });
         gameTimer.start();
 
-        setVisible(true);
     }
 
-    private void initializeBricks() {
-        Random random = new Random();
-        int xoffset = (getWidth() - (COLS * (BRICK_WIDTH + SPACING) - SPACING)) / 2;
+    public BrickBreakerFrame(NeuralNetwork bestNw) {
```

Comment:
Maybe you can reduce the amount of code duplication with these two constructors?